pkg/controller/pod.go

```diff
@@ -202,15 +214,87 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		}
 	}
 
-	klog.V(4).Infof("Registering instance %s under service %+v", pod.Name, cloudmapConfig)
+	var serviceItem *CloudMapInstanceCacheItem
+	serviceInstanceSummary := make(map[string]bool)
+	cloudMapServiceKey := *cloudmapConfig.NamespaceName + "-" + *cloudmapConfig.ServiceName
+
+	existingItem, exists, _ := c.cloudMapInstanceCache.Get(&CloudMapInstanceCacheItem{
+		key: cloudMapServiceKey,
+	})
+
+	if exists && existingItem !=nil {
+		serviceItem = existingItem.(*CloudMapInstanceCacheItem)
```

Comment:
i still don't understand how could (existingItem == nil) if (exists == true) (since we are always put a valid non-nil item into cache). Is it just double check to make things work? (or bypassed some hidden bugs)